vms/administrator/tests/test_settings.py

```diff
@@ -342,31 +660,198 @@ def test_edit_job(self):
+            'changed job name')
+        self.driver.find_element_by_xpath('//textarea[@name = "description"]').clear()
+        self.driver.find_element_by_xpath('//textarea[@name = "description"]').send_keys(
+            'changed-job-description')
+        self.driver.find_element_by_xpath('//input[@name = "start_date"]').clear()
+        self.driver.find_element_by_xpath('//input[@name = "start_date"]').send_keys(
+            '09/01/2017')
+        self.driver.find_element_by_xpath('//input[@name = "end_date"]').clear()
+        self.driver.find_element_by_xpath('//input[@name = "end_date"]').send_keys(
+            '09/11/2017')
+
+        self.driver.find_element_by_xpath('//form[1]').submit()
+
+        # check job not edited and proper error message displayed
+        self.assertNotEqual(self.driver.current_url, self.live_server_url + '/job/list/')
+        self.assertEqual(self.driver.find_element_by_xpath('//div[2]/div[3]/p').text,
+            'You cannot edit this job as 1 associated shift no longer lies within the new date range')
+
+        self.driver.get(self.live_server_url + '/job/list/')
+
+        # edit job with date range such that the shift start date no longer falls in the range
+        self.assertEqual(self.driver.find_element_by_xpath('//table//tbody//tr[1]//td[6]').text,
+            'Edit')
+        self.driver.find_element_by_xpath('//table//tbody//tr[1]//td[6]//a').click()
+
+        # edit job with date range such that the shift end date no longer falls in the range
+        self.driver.find_element_by_xpath('//input[@name = "name"]').clear()
+        self.driver.find_element_by_xpath('//input[@name = "name"]').send_keys(
+            'changed job name')
+        self.driver.find_element_by_xpath('//textarea[@name = "description"]').clear()
+        self.driver.find_element_by_xpath('//textarea[@name = "description"]').send_keys(
+            'changed-job-description')
+        self.driver.find_element_by_xpath('//input[@name = "start_date"]').clear()
+        self.driver.find_element_by_xpath('//input[@name = "start_date"]').send_keys(
+            '08/22/2017')
+        self.driver.find_element_by_xpath('//input[@name = "end_date"]').clear()
+        self.driver.find_element_by_xpath('//input[@name = "end_date"]').send_keys(
+            '08/26/2017')
+
+        self.driver.find_element_by_xpath('//form[1]').submit()
+
+        # check job not edited and proper error message displayed
+        self.assertNotEqual(self.driver.current_url, self.live_server_url + '/job/list/')
+        self.assertEqual(self.driver.find_element_by_xpath('//div[2]/div[3]/p').text,
+            'You cannot edit this job as 1 associated shift no longer lies within the new date range')
+
     def test_delete_job_without_associated_shift(self):
         self.login_admin()
 
         # register event first to create job
-        event = ['event-name', '08/21/2016', '09/28/2016']
+        event = ['event-name', '08/21/2017', '09/28/2017']
```

Comment:
@tapasweni-pathak Yes, for event the start date has to be later than today. I had to change dates for a few functions, where the dates were set in June and then I went ahead and changed it in most places to a date later in the future so that I don't face build failures as and when the date elapses.
Later once, I put in helper functions for all files and restructure the code, these dates would have to be changed in only 1 place and I will put in logic such that the generated date is later than today